crates/binjs_io/src/entropy/rw.rs

```diff
@@ -174,16 +175,37 @@ pub struct TableRefStreamState<T> {
     /// Phantom type, to ensure that we only ever use a given `TableRefStreamState` with
     /// a single (type of) LinearTable.
     phantom: PhantomData<T>,
+
+    /// The latest references encountered.
+    ///
+    /// window[0] is the latest reference encountered,
+    /// window[1] the second latest
+    /// ...
+    ///
+    /// Invariant:
+    /// - length is always <= window_max_len
+    /// - if i ≠ j, window[i] ≠ window[j]
+    window: Vec<TableRef>,
+
+    /// The maximal length of `self.window`, which is also the number of
+    /// integers to reserve to represent backreferences to values encountered
+    /// recently.
+    window_max_len: usize,
 }
 impl<T> TableRefStreamState<T>
 where
     T: Eq + std::hash::Hash + Clone,
 {
     /// Create a new `TableRefStreamState`.
-    pub fn new() -> Self {
+    ///
+    /// `window_max_len` represents the number of integers to reserve to
+    /// represent backreferences to values encountered recently.
+    pub fn new(window_max_len: usize) -> Self {
         TableRefStreamState {
             latest_in_prelude: None,
             phantom: PhantomData,
+            window_max_len,
```

Comment:
Do we need this as a separate field or is it essentially same as `window.capacity()`?

Reply:
It's essentially the same as `window.capacity()`, but keeping them separate lets me assert that I'm not doing something stupid that will accidentally extend `window.capacity()`.

Do you think I should hide this behind a `#[cfg(debug_assertions)]`?

Reply:
Hmm fair enough. I think this is fine as-is then.